spec/client/permissions_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'spec_helper'
+
+describe Elastic::EnterpriseSearch::Client::Permissions do
+  let(:client) { Elastic::EnterpriseSearch::Client.new }
+
+  before :each do
+    Elastic::EnterpriseSearch.access_token = 'xyT3mm3ecPsPuYxd_6fX'
+  end
+
+  let(:content_source_key) { '5d96387b75444bf49532c24a' }
+  let(:user) { 'enterprise_search' }
+
+  context '#list_all_permissions' do
+    it 'lists all permissions' do
+      VCR.use_cassette(:list_all_permissions) do
+        response = client.list_all_permissions(content_source_key)
+        expect(response['results'].size).to eq(2)
+      end
+    end
+
+    it 'lists all permissions with paging' do
+      VCR.use_cassette(:list_all_permissions_with_paging) do
+        response = client.list_all_permissions(content_source_key, :current => 2, :size => 5)
+        expect(response['meta']['page']['current']).to eq(2)
+        expect(response['meta']['page']['size']).to eq(5)
+        expect(response['results'].size).to eq(0)
+      end
+    end
+  end
+
+  context '#add_user_permissions' do
+    let(:permissions) { ['permission1'] }
```

Comment:
Do you think it's better to ask users pass permissions body as it required in API and not wrap it in object for them?

So instead of passing ['permission1'] they will pass {permissions: ['permission1']}.

I'm asking, because I didn't know what's the right path here and went with wrapping it in object inside the client: https://github.com/elastic/enterprise-search-node/blob/master/lib/enterpriseSearch.js#L58

Reply:
@yakhinvadim Haha, that's funny. I also wasn't sure what to do so I did what you did. I really don't know what the correct answer here is.

One one hand, if we just accept the body we'll never have to touch this method again, even if new parameters are added. Additionally, we can defer to the API docs for the body specification.

On the other hand, it does seem slightly simpler as a user to just pass permissions array.

Reply:
Let's go with not wrapping?
I think future-proofing is more valuable than "slightly simpler"

Reply:
https://github.com/elastic/enterprise-search-ruby/pull/15/commits/e0caae1d310f5afa1dad638da7ed88edcb0c4ecb